pkg/controller/namespace/namespace_controller.go

```diff
@@ -132,15 +134,15 @@ func (nm *NamespaceController) enqueueNamespace(obj interface{}) {
 // Each namespace can be in the queue at most once.
 // The system ensures that no two workers can process
 // the same namespace at the same time.
-func (nm *NamespaceController) worker() {
-	workFunc := func() bool {
+func (nm *NamespaceController) worker() func(context.Context) {
```

Comment:
Any particular reason for this approach rather than accepting context as a parameter to worker? That way the function will be simple than what is proposed here.

Reply:
I agree, that should also work.

Reply:
ok, I have updated it